array_schema.go

```diff
@@ -11,9 +11,54 @@ import (
 	"fmt"
 	"os"
 	"runtime"
+	"sync/atomic"
 	"unsafe"
 )
 
+type arraySchemaHandle struct {
+	// An unsafe pointer to the capiHandle holding the schema.
+	// This pointer must be accessed using atomic functions, because the MarshalJSON function
+	// replaces the native handle of an existing object.
+	// We cannot wrap atomic.Pointer[capiHandle] in a struct and pass it around because of lint warnings,
+	// and we cannot mutate the native handle of an existing capiHandle because it's impossible to atomically
+	// set the finalizer.
+	h unsafe.Pointer
```

Comment:
Could this struct hold any capiHandle since it's only member is an `unsafe.Pointer`?  I'm noticing it seems not specific to the array schema aside from definitions for `Get` and `Reset`, and it feels like other TileDB-Go types would do something similar and wrap `capiHandle` with `unsafe.Pointer`?

Reply:
No worries, feel free to disregard - I'm probably thinking too much like C++. If ArraySchemaHandle is the only type that needs to use atomics as you mentioned in your reply to my other comment then there's really no point to thinking about a base class anyhow.